.github/workflows/scripts/registry-login.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+#
+# Copyright 2023 SLSA Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set -euo pipefail
+
+image_or_repo="${1}"
+username="${2}"
+password="${3}"
+gcp_access_token="${4}"
+
+default_registry="docker.io"
+
+# NOTE: Some docker images are of the form <org>/<name>
+# Here we get the first part and check if it has a '.' or ':'
+# character in it to see if it's a domain name.
+# See: https://stackoverflow.com/questions/37861791/how-are-docker-image-names-parsed#37867949
+maybe_domain=$(echo "${image_or_repo}" | cut -f1 -d "/" | grep -E "\.|:" || true)
```

Comment:
I think the your new behavior would be equivalent, but let's copy-paste to be sure.
```suggestion
maybe_domain=$(echo "${image_or_repo}" | cut -f1 -d "/" | { grep -E "\.|:" || true; })
```